### PR TITLE
mds,qa: some balancer debug messages (<=5) not printed when debug_mds is >=5

### PR DESCRIPTION
**Pull request by batrick**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
Note, it's not easy to really fix this unique debugging quirk in the MDS. I chose to leave it this way and bump debugging in the qa suite.

One option to "fix" this for most who would otherwise trip on this is to change the debug_mds_balancer default to `1/1`.

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2022-06-12_05:08:12-fs:workload-wip-pdonnell-testing-20220612.004943-distro-default-smithi/

This is good to merge.

**Review by lxbsz — APPROVED** [review]
Nice catch and LGTM.

File: src/mds/MDBalancer.cc [hunk 1]
```diff
@@ -37,6 +37,10 @@ using namespace std;
 #include "common/config.h"
 #include "common/errno.h"
 
+/* Note, by default ceph_mds_balancer is 1/5. For debug messages 1<lvl<=5, should_gather will
+ * be true so whatever debug_mds is is ignored. So, some messages may not appear in the log.
```
Comment (dparmar18) [inline src/mds/MDBalancer.cc:41]: ".....be true so whatever debug_mds is, it is ignored. So, some messages may not appear in the log."

Just an opinion: s/is is/is, it is

**Comment by dparmar18** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2022-June-13

Good to merge once jenkins passes.

**Review by dparmar18 — APPROVED** [review]

**Comment by dparmar18** [comment]
LGTM

**Review by ajarr — APPROVED** [review]

**Review by vshankar — APPROVED** [review]

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-July-04

**Pull request merged**